### ISSUE TITLE
Improve post-report messaging

### ISSUE THREAD

**Issue by garethrees** [mysociety/alaveteli#7778]
One of the issues raised in https://github.com/mysociety/whatdotheyknow-private/issues/219 was whether we should auto acknowledge reports. This would get in to some annoying details with threading and we guess this is probably on the lower end of impact.

What we could do fairly easily is improve the text that appears when something is reported.

Currently you fill out the form…

<img width="1074" alt="Screenshot 2023-06-19 at 09 45 27" src="https://github.com/mysociety/alaveteli/assets/282788/09406242-2146-4a53-b5b6-1920b1a2b2c4">

…and on submission just get a basic success flash…

<img width="1125" alt="Screenshot 2023-06-19 at 09 49 06" src="https://github.com/mysociety/alaveteli/assets/282788/b6966c26-29f4-4862-9710-71b06ca78c2e">

We could add an overridable partial to that flash, so that site runners can either add more to the message (a list of bullet points of the process admins will follow) and/or a link to a help page that describes the process in more detail.

We'll want to make this generic in alaveteli, and then add some WDTK specific content.

Bonus points if we have a `@report` object available that details the particular report made that we could use to dynamically pick the content to render (as in something like…)

```erb
<%=
   case @report.reason
   when 'defamatory'
     render partial: 'report/next_steps/defamatory'
   when 'vexatious'
      render partial: 'report/next_steps/vexatious'
   else
      render partial: 'report/next_steps/general'
%>
```

**Comment by mdeuk** [comment]
> One of the issues raised in https://github.com/mysociety/whatdotheyknow-private/issues/219 was whether we should auto acknowledge reports.

You must be a mind reader, as this was something I pondered over the weekend.

If this is something we _want_ to do, we can do it once the Monster handles labelling of inbound mail. I would envision this being a generic response, rather than one that quoted the report itself [^1].

<details>
<summary>Early draft copy</summary>

An early draft of what this _could_ look like. It would need refinement.

> _**Case reference:** XXX_
> 
> _Thank you for taking the time to make us aware of your concerns._
> 
> _Our team will review the information you've provided, along with the request itself, and get back to you in due course._
> 
> _For more information about how we handle these cases, please see https://www.whatdotheyknow.com/help/how#what_happens_

</details>

Related: 
- https://github.com/mysociety/whatdotheyknow-theme/issues/740#issuecomment-1581235710

[^1]: so as to avoid backscatter containing PII. We would probably also want to send from `DoNotReply@`.

**Comment by garethrees** [comment]
🕴️ 

Yeah, I think auto-acks _feels_ more the responsibility of the support system than Alaveteli, though I'm not sure I could fully back up my reasoning for that.

I think let's see where improving the Alaveteli UI gets us, and then ticket up auto-acks for the monster if it still feels we want it.

**Comment by mdeuk** [comment]
> Yeah, I think auto-acks feels more the responsibility of the support system than Alaveteli

Yup, I agree… it's certainly how you would do it with a ticketing system, so for re-user compatibility, I think 'outsourcing' the function back to whatever runs their support "system" would be better.

**Comment by garethrees** [comment]
We've got some good content about how we deal with these reports at https://www.whatdotheyknow.com/help/how#what_happens. Here's how I think they map to the form field values.

| Form value                       | Relevant help section                                                                                                 |
|----------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| Contains defamatory material     | https://www.whatdotheyknow.com/help/how#potentially_libellous                                                         |
| Not a valid request              | https://www.whatdotheyknow.com/help/how#invalid_requests                                                              |
| Request for personal information | https://www.whatdotheyknow.com/help/how#invalid_requests                                                              |
| Contains personal information    | https://www.whatdotheyknow.com/help/how#accidentally_released<br>– a bit vague that PII can be intentionally released |
| Vexatious                        | https://www.whatdotheyknow.com/help/how#vexatious_requests                                                            |
| Other                            | https://www.whatdotheyknow.com/help/how#copyright,<br>https://www.whatdotheyknow.com/help/how#compassionate_grounds   |

I expect many people report content having never seen this guidance.

Perhaps instead of linking to this in the _post_ report flash, we render the guidance specific to the currently selected field on the new report form page.

![Screenshot 2023-07-04 at 10 30 01](https://github.com/mysociety/alaveteli/assets/282788/8c49eee2-49b1-4146-a9b1-71f38d6eb8e3)

We could extract each section to a partial (`report/how/vexatious` – similar to above) and render if it exists, or fall back to the generic message.

**Comment by HelenWDTK** [comment]
We could do both. Render the guidance on the report form, then send a generic acknowledgement to confirm that the report has been received.

**Comment by garethrees** [comment]
Just noting here that we have a help section related to reporting at https://www.whatdotheyknow.com/help/about#reporting_unavailable
